File: 2023/Day 1/1-1.py
input = open("Day 1/1-1.txt")

list = []

for line in input.readlines():
    first = False
    last = False

    for char in line:
        if char.isnumeric():
            if not first:
                first = char
            last = char

    list.append(int(first + last))

print(list)
print(sum(list))

File: 2024/Day 1/1-1.ts
import * as fs from 'fs';

const input = fs.readFileSync('./Day 1/input.txt', 'utf-8');
const lines = input.split(/\r?\n/);

function main(){
  const leftList: number[] = [];
  const rightList: number[] = [];
  let distance = 0
  
  lines.forEach((line) => {
    const list = line.split(/\s+/).map(Number);
    leftList.push(list[0]);
    rightList.push(list[1]);
  });
  leftList.sort();
  rightList.sort();

  for(let i = 0; i < lines.length; i++){
    distance += Math.abs(leftList[i] - rightList[i]);
  }

  console.log(distance);
}

main()